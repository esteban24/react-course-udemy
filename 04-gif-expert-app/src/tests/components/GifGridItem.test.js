import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('GifGridItem.js test suite', () => {
    const title = "Sample Text";
    const url = "https://my.test.host/gif.gif";
    const wrapper = shallow(<GifGridItem url={ url } title={ title }/>);

    test('should match component snapshop', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should have a <p></p> and match title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('should have <img/> with src and alt properties', () => {
        const img = wrapper.find('img');

        expect(img.prop('alt')).toBe(title);
        expect(img.prop('src')).toBe(url);
    })

    test('should have card animate animate__fadeIn class', () => {
        const div = wrapper.find('div');
        
        expect(div.prop('className')).toBe("card animate__animated animate__fadeIn");
    })
})
