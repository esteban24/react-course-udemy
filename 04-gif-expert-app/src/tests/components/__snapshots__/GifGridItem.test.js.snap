// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem.js test suite should match component snapshop 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Sample Text"
    src="https://my.test.host/gif.gif"
  />
  <p>
    Sample Text
  </p>
</div>
`;
